contrib/gogearbox/gearbox.v1/internal/gearboxutil/fasthttpcarrier_test.go

```diff
@@ -0,0 +1,76 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016 Datadog, Inc.
+
+package gearboxutil
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/valyala/fasthttp"
+)
+
+func TestFasthttpCarrierSet(t *testing.T) {
+	assert := assert.New(t)
+	fcc := &FasthttpCarrier{
+		ReqHeader: new(fasthttp.RequestHeader),
+	}
+
+	// add one item
+	fcc.Set("k1", "v1")
+	assert.Len(fcc.ReqHeader.PeekAll("k1"), 1)
+	assert.Equal("v1", string(fcc.ReqHeader.Peek("k1")))
+
+	// // add a second value
+	fcc.Set("k1", "v2")
+	vals := fcc.ReqHeader.PeekAll("k1")
+	assert.Len(vals, 2)
+	assert.Equal("v1", string(vals[0]))
+	assert.Equal("v2", string(vals[1]))
+
+	// // add a second item
+	fcc.Set("k2", "v21")
+	// assert.Len(fcc.ReqHeader.PeekKeys(), 2) This returns 3 values: k1, k1 and k2...
```

Comment:
```suggestion
	 assert.Len(fcc.ReqHeader.PeekKeys(), 2)
```

I'm not sure why fasthttp has this behaviour, but it should be ok after updating the `Set` implementation to override instead of appending.

Reply:
Leaving this open given the unresolved discussion on fasthttpcarrier.Set method